src/random.h

```diff
@@ -25,6 +26,29 @@ int GetRandInt(int nMax);
 uint256 GetRandHash();
 
 /**
+ * Rearranges the elements in the range [first,first+len) randomly, assuming
+ * that gen is a uniform random number generator. Follows the same algorithm as
+ * std::shuffle in C++11 (a Durstenfeld shuffle).
```

Comment:
[comment] I'm curious why we don't use ``std::shuffle`` if we're reimplementing the same algorithm. If we ``std::shuffle`` a list of pairs of ``(i, x)`` where ``i`` is the original index, could we achieve the same result while reusing ``std``? (I can't tell if that'd be nicer or worse for ergonomics, yet.)